error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -1883,6 +1945,30 @@ Duration after(StepVerifier.LastStep step, Consumer<Throwable> consumer) {
     }
   }
 
+  /**
+   * Prefer {@link StepVerifier.LastStep#verifyErrorSatisfies(Consumer)} with AssertJ over more
+   * contrived alternatives.
+   */
+  static final class StepVerifierLastStepVerifyErrorSatisfiesAssertJ {
+    @BeforeTemplate
+    void before(StepVerifier.LastStep step, Class<? extends Throwable> clazz, String message) {
+      Refaster.anyOf(
+          step.expectError()
+              .verifyThenAssertThat()
+              .hasOperatorErrorOfType(clazz)
+              .hasOperatorErrorWithMessage(message),
+          step.expectError(clazz).verifyThenAssertThat().hasOperatorErrorWithMessage(message),
+          step.expectErrorMessage(message).verifyThenAssertThat().hasOperatorErrorOfType(clazz));
+    }
+
+    @AfterTemplate
+    @UseImportPolicy(STATIC_IMPORT_ALWAYS)
+    void after(StepVerifier.LastStep step, Class<? extends Throwable> clazz, String message) {
+      step.verifyErrorSatisfies(
+          t -> Assertions.assertThat(t).isInstanceOf(clazz).hasMessage(message));
+    }
+  }
```

Comment:
Asserting both class and message blocks one to use the recommended `StepVerifier.Step#verifyError(Class)` or `StepVerifier.Step#verifyErrorMessage(String)`. Choosing an alternative isn't easy since there are many possibilities. Out of all the options, AssertJ's assertions have the richest API, so I'd recommend this.

The `StepVerifierVerify` rule introduced in this PR aims to rewrite the use-cases in which only one of class or message are asserted by dropping the `StepVerifier.Assertions` type which allows other rules to rewrite further.